support-frontend/assets/helpers/tracking/acquisitions.js

```diff
@@ -283,16 +274,54 @@ function deserialiseReferralData(serialised: string): Object {
   };
 }
 
+
+// Reads the acquisition data from sessionStorage.
+function getReferrerAcquisitionDataFromSessionStorage(): ?ReferrerAcquisitionData {
+  const stored = storage.getSession(ACQUISITIONS_STORAGE_KEY);
+  return stored ? deserialiseJsonObject(stored) : null;
+}
+
+// Reads the acquisition data from the &acquistionData param containing a serialised JSON string.
+function getAcquisitionDataFromAcquisitionDataParam(): ?Object {
+  if (getQueryParameter(ACQUISITIONS_PARAM)) {
+    return deserialiseJsonObject(getQueryParameter(ACQUISITIONS_PARAM) || '');
+  }
+  return null;
+}
+
+// Reads the acquisition data from the &referralData param containing _ separated values.
+function getAcquisitionDataFromReferralDataParam(): ?Object {
+  if (getQueryParameter(REFERRAL_DATA_PARAM)) {
+    return deserialiseReferralData(getQueryParameter(REFERRAL_DATA_PARAM) || '');
+  }
+  return null;
+}
+
+// Generates appropriate acquisition data from the various, known, PPC params
+function getAcquisitionDataFromPPCParams(): ?Object {
+  if (getQueryParameter('gclid')) {
+    return {
+      source: 'PPC',
+      campaignCode: 'Google_Adwords',
+      queryParameters: getAllQueryParams(),
```

Comment:
this line caused a backend serialisation error for users with this kind of ID and has been fixed in this PR https://github.com/guardian/support-frontend/pull/3222